Frontend/ui-customer/src/Screens/ParcelDeliveryFlow/ParcelDeliveryScreen/ComponentConfig.purs

```diff
@@ -71,6 +82,124 @@ genericHeaderConfig state = let
     }
   in genericHeaderConfig'
 
+deliveryPickupDetialsModalConfig :: ST.ParcelDeliveryScreenState -> PopUpModal.Config
+deliveryPickupDetialsModalConfig state = 
+  let
+      config' = PopUpModal.config
+      config = getAppConfig appConfig
+      isSenderModal = state.data.currentStage == ST.SENDER_DETAILS
+      details = if isSenderModal then state.data.senderDetails else state.data.receiverDetails
```

Comment:
create a function for this which would take isSenderModal as input and return an object.